cppcache/src/SerializationRegistry.hpp

```diff
@@ -320,8 +334,12 @@ class APACHE_GEODE_EXPORT SerializationRegistry {
 
   inline void serialize(const DataSerializable* obj, DataOutput& output,
                         bool isDelta) const {
-    auto id = obj->getClassId();
-    auto dsCode = getSerializableDataDsCode(id);
+    auto&& type = obj->getType();
+    auto typeIterator = theTypeMap.typeToClassId.find(type);
+    auto id = typeIterator->second;
+
+    typeid(*obj).name();
+    auto dsCode = getSerializableDataDsCode((int32_t)id);
```

Comment:
If `DataSerializable` class ID is now `int32_t` then the call to getting the DSCode to encode the ID shouldn't need any casting right?